tika-parsers/src/main/java/org/apache/tika/parser/crypto/TSDParser.java

```diff
@@ -0,0 +1,361 @@
+//package it.notartel.cons.verifyfiles.custom;
+package org.apache.tika.parser.crypto;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.mime.MediaType;
+import org.apache.tika.parser.AbstractParser;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.sax.XHTMLContentHandler;
+import org.bouncycastle.asn1.cryptopro.CryptoProObjectIdentifiers;
+import org.bouncycastle.asn1.nist.NISTObjectIdentifiers;
+import org.bouncycastle.asn1.oiw.OIWObjectIdentifiers;
+import org.bouncycastle.asn1.pkcs.PKCSObjectIdentifiers;
+import org.bouncycastle.asn1.teletrust.TeleTrusTObjectIdentifiers;
+import org.bouncycastle.asn1.x509.GeneralName;
+import org.bouncycastle.asn1.x509.X509ObjectIdentifiers;
+import org.bouncycastle.asn1.x9.X9ObjectIdentifiers;
+import org.bouncycastle.cms.CMSSignedDataGenerator;
+import org.bouncycastle.tsp.TimeStampToken;
+import org.bouncycastle.tsp.cms.CMSTimeStampedDataParser;
+//import org.slf4j.Logger;
+//import org.slf4j.LoggerFactory;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.SAXException;
+
+/*
+
+Nome Formato:            Time Stamped Data Envelope
+
+Mime Type:               application/timestamped-data
+
+Estensione:              .tsd
+
+*/
+public class TSDParser extends AbstractParser {
+	
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 6139181424595882376L;
+
+	//private static final Logger log = LoggerFactory.getLogger(TSDParser.class);
```

Comment:
`private static final Logger LOG = LoggerFactory.getLogger(TSDParser.class);`

We haven't updated info for contributors but it would be preferred signature for logger.